src/test/java/ru/spbstu/calculator/PWCalculatorTest.java

```diff
@@ -1,55 +1,59 @@
-
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('A').getRow().length);
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('C').getRow().length);
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('G').getRow().length);
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('T').getRow().length);
-
-        assertEquals(0.1823215567939546D, matrix.getRowByNucleotide('A').getByIndex(0), delta);
-        assertEquals(-0.2231435513142097D, matrix.getRowByNucleotide('C').getByIndex(1), delta);
-        assertEquals(1.0296194171811581D, matrix.getRowByNucleotide('G').getByIndex(2), delta);
-        assertEquals(Double.NEGATIVE_INFINITY, matrix.getRowByNucleotide('T').getByIndex(3), delta);
-
-        strategy = new CalculationStrategy(frequencyOfNucleotides, sequenceLength, relative);
-        matrix = pwcalculator.calculateMatrix(fastaFile, strategy);
-
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('A').getRow().length);
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('C').getRow().length);
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('G').getRow().length);
-        assertEquals(sequenceLength, matrix.getRowByNucleotide('T').getRow().length);
-
-        assertEquals(0.3D, matrix.getRowByNucleotide('A').getByIndex(0), delta);
-        assertEquals(0.2D, matrix.getRowByNucleotide('C').getByIndex(1), delta);
-        assertEquals(0.5D, matrix.getRowByNucleotide('G').getByIndex(7), delta);
-        assertEquals(0.6D, matrix.getRowByNucleotide('T').getByIndex(8), delta);
-    }
-}
+package ru.spbstu.calculator;
+
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import java.io.IOException;
+
+import ru.spbstu.fastafile.FastaFile;
+import ru.spbstu.calculator.Constants.calculationMethod;
+
+public class PWCalculatorTest {
+    private FastaFile fastaFile;
+    private final String fileName = "src/test/resources/PWTest";
+    private final double delta = 0.00000000001;
+    private int sequenceLength = 9;
+
+    @Before
+    public void setup() throws IOException {
+        fastaFile = new FastaFile();
+        fastaFile.readFastaFile(fileName);
+    }
+
+    @Test
+    public void testLOGLIKELIHOOD() {
+        PWCalculator pwCalculator = new PWCalculator();
+        PWMatrix matrix = pwCalculator.calculateMatrix(fastaFile, new CalculationStrategy.Builder()
+                .sequenceLength(sequenceLength)
+                .build());
+
+        assertEquals(sequenceLength, matrix.getRowByNucleotide('A').getRow().length);
+        assertEquals(sequenceLength, matrix.getRowByNucleotide('C').getRow().length);
+        assertEquals(sequenceLength, matrix.getRowByNucleotide('G').getRow().length);
+        assertEquals(sequenceLength, matrix.getRowByNucleotide('T').getRow().length);
+
+        assertEquals(0.1823215567939546D, matrix.getRowByNucleotide('A').getByIndex(0), delta);
```

Comment:
Does `Double.compare()` fit here?

Reply:
`assertEquals` has implementation with Doubles: https://junit.org/junit4/javadoc/4.12/org/junit/Assert.html#assertEquals(double,%20double,%20double)
Did you mean using `assertThat(Double.compare(a, b)).isTrue()`? Is this not redundant?